apollo-router/build/spaceport.rs

```diff
@@ -23,30 +20,16 @@ pub fn main() -> Result<(), Box<dyn Error>> {
     content = content.replace("[(js_use_toArray)=true]", "");
     content = content.replace("[(js_preEncoded)=true]", "");
 
+    let out_dir = PathBuf::from(std::env::var_os("OUT_DIR").unwrap());
     let src = PathBuf::from(std::env::var_os("CARGO_MANIFEST_DIR").unwrap()).join("src");
     let proto_dir = src.join("spaceport").join("proto");
-    // Try to avoid writing out the same content since it will trigger unnecessary re-builds, which wastes time
-    let write_content = match File::open(proto_dir.join("reports.proto")) {
-        Ok(mut existing) => {
-            let mut existing_content = String::new();
-            existing.read_to_string(&mut existing_content)?;
-            content != existing_content
-        }
-        Err(_) => true,
-    };
+    let agents = proto_dir.join("agents.proto");
+    let reports = out_dir.join("reports.proto");
 
-    // Write the content out if they differ or an error occured trying to open proto file
-    if write_content {
-        let mut dest = File::create(proto_dir.join("reports.proto"))?;
-        copy(&mut content.as_bytes(), &mut dest)?;
-    }
+    std::fs::write(&reports, &content)?;
 
     // Process the proto files
-    let proto_files = vec![
-        proto_dir.join("agents.proto"),
-        proto_dir.join("reports.proto"),
-    ];
-
```

Comment:
Ah, now I remember why I couldn't put it into OUT_DIR. This was a little trick to try and avoid annoying re-compilations when nothing has really changed. Is there a way to preserve this or is it not actually required now?

Reply:
Indeed, this PR caused a rebuild every time. I’ve removed `cargo:rerun-if-changed` for `reports.proto` to fix it.

It means we won’t automatically recompile if https://usage-reporting.api.apollographql.com/proto/reports.proto has changed, but that was effectively already the case.

Reply:
From discussion with Gary: I’ll make a follow-up PR to add `reports.proto` to the source repository, have the build script use that without downloading, add an xtask command to download and update the file, and add a test that downloads and checks our copy is up to date.